HACKING.md

```diff
@@ -247,6 +244,28 @@ In a Debian/Ubuntu build environment you can build debs with
 
     tools/make-debs --quick
 
+## Updating `node_modules`
+
+During a normal build from a git checkout, the `node_modules` will be
+automatically unpacked from a cache kept in a separate git repository.  You can
+force the unpack to occur using the `tools/node-modules checkout` command, but
+this shouldn't be necessary.  In the event that you need to modify
+`package.json` (to install a new module, for example) then you'll need to run
+`tools/node-modules rebuild` to create a new cache from the result of running
+`npm install` on your new `package.json`.
+
+Before landing changes to the main cockpit project (or proposing a PR), this
+rebuild will need to be pushed to the cache repository.  `tools/node-modules
+push` will do this for you.
+
+The `tools/node-modules` script inspects the `GITHUB_BASE` environment variable
+to determine the correct repository to use when fetching and pushing.  It will
+strip the repository name (leaving the project- or username) and use the
+`node-cache.git` repository in that namespace.  If `GITHUB_BASE` is unset, it
+will default to `cockpit-project/node-cache.git`.
+
+A local cache is maintained in `~/.cache/cockpit-dev/node-cache.git`.
```

Comment:
Remember to adjust that according to the "needs github org in the path" comment from above.

Reply:
... does this even need to be documented that explicitly? Perhaps it's enough to just say `~/.cache/cockpit-dev/` here? It's not like we want people to start making assumptions about it, it should remain a private implementation detail IMHO.